dll/win32/shell32/wine/pidl.c

```diff
@@ -485,6 +485,37 @@ LPITEMIDLIST WINAPI ILGlobalClone(LPCITEMIDLIST pidl)
     return newpidl;
 }
 
+#ifdef __REACTOS__
+static inline LPITEMIDLIST _ILUnsafeNext(LPCITEMIDLIST pidl)
+{
+    return (LPITEMIDLIST)(((BYTE*)pidl) + pidl->mkid.cb);
+}
+
+UINT _ILGetDepth(LPCITEMIDLIST pidl)
+{
+    for (UINT i = 0;; ++i)
+    {
+        if (!pidl || !pidl->mkid.cb)
+            return i;
+        pidl = _ILUnsafeNext(pidl);
+    }
+}
+
+static BOOL _ILMemCmpEqualIDList(LPCITEMIDLIST p1, LPCITEMIDLIST p2)
+{
+    for (;; p1 = _ILUnsafeNext(p1), p2 = _ILUnsafeNext(p2))
+    {
+        DWORD cb1 = p1 ? p1->mkid.cb : 0x80000000; /* Empty != NULL */
+        DWORD cb2 = p2 ? p2->mkid.cb : 0x80000000;
```

Comment:
Is the `0x80000000` a flag of sorts?

Reply:
I guess you can call it that, it just needs to be different from `(WORD)0`. I don't know why Windows `ILIsEqual` feels like empty != NULL.

We are already testing this in apitests: `ok_int(ILIsEqual(NULL, &emptyitem), FALSE);`

Reply:
Ah ok, so some sort of "dummy" value.